modules/tinymce/src/plugins/image/test/ts/browser/ImagePluginTest.ts

```diff
@@ -1,196 +1,223 @@
-      Chain.asStep({}, [
-        cFillActiveDialog(data, true)
-      ]),
-      ui.sClickOnUi('click save', 'div[role="dialog"] button:contains("Save")'),
-      api.sAssertContent(expectedContent)
-    ]);
-
-    const createTestOnEmptyEditor = (name: string, data: Partial<ImageDialogData>, expectedContent: string) => createTestWithContent(name, '', { elementPath: [ 0 ], offset: 0 }, data, expectedContent);
-
-    const createTestUpdatedStyle = (name: string, style: string, assertion: Step<any, any>) => Log.stepsAsStep('TBA', 'Image: ' + name, [
-      sInitAndOpenDialog('', { elementPath: [ 0 ], offset: 0 }),
-      ui.sClickOnUi('Switch to Advanced tab', '.tox-tab:contains("Advanced")'),
-      Chain.asStep(SugarBody.body(), [
-        cSetInputValue(advancedTabSelectors.style, style)
-      ]),
-      assertion,
-      ui.sClickOnUi('click save', 'div[role="dialog"] button:contains("Save")')
-    ]);
-
-    const suiteArr = [
-      createTestOnEmptyEditor(
-        'Advanced image dialog margin space options on empty editor',
-        {
-          alt: 'alt',
-          hspace: '10',
-          src: {
-            value: 'src'
-          },
-          vspace: '10'
-        },
-        '<p><img style="margin: 10px;" src="src" alt="alt" /></p>'
-      ),
-      createTestOnEmptyEditor(
-        'Advanced image dialog border style only options on empty editor',
-        {
-          alt: 'alt',
-          src: {
-            value: 'src'
-          },
-          style: 'border-width: 10px; border-style: solid;'
-        },
-        '<p><img style="border-width: 10px; border-style: solid;" src="src" alt="alt" /></p>'
-      ),
-      createTestOnEmptyEditor(
-        'Advanced image dialog margin style only options on empty editor',
-        {
-          alt: 'alt',
-          src: {
-            value: 'src'
-          },
-          style: 'margin: 10px;'
+import Theme from 'tinymce/themes/silver/Theme';
+
+import {
+  advancedTabSelectors, assertInputValue, fillActiveDialog, ImageDialogData, pWaitForDialog, setInputValue, submitDialog
+} from '../module/Helpers';
+
+describe('browser.tinymce.plugins.image.ImagePluginTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'image',
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ]);
+
+  const pInitAndOpenDialog = async (editor: Editor, content: string, cursorPos: Cursors.CursorSpec | Cursors.RangeSpec) => {
+    editor.settings.image_advtab = true;
+    editor.settings.image_dimensions = false;
+    editor.setContent(content);
+    TinySelections.setSelectionFrom(editor, cursorPos);
+    editor.execCommand('mceImage');
+    await pWaitForDialog(editor);
+  };
+
+  const pCreateTestWithContent = async (editor: Editor, content: string, cursorPos: Cursors.CursorSpec | Cursors.RangeSpec, data: Partial<ImageDialogData>, expectedContent: string) => {
+    await pInitAndOpenDialog(editor, content, cursorPos);
```

Comment:
Do we want to keep this in a context also to mirror the logging from before?

Reply:
No, the logging before was for substeps within a single test. There's no way to do that in the new system and it's honestly not needed.